hedera-node/src/main/java/com/hedera/services/queries/token/GetAccountNftInfosAnswer.java

```diff
@@ -0,0 +1,180 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.context.primitives.StateView;
+import com.hedera.services.queries.AnswerService;
+import com.hedera.services.txns.validation.OptionValidator;
+import com.hedera.services.utils.SignedTxnAccessor;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenGetAccountNftInfosQuery;
+import com.hederahashgraph.api.proto.java.TokenGetAccountNftInfosResponse;
+import com.hederahashgraph.api.proto.java.TokenNftInfo;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.hedera.services.utils.SignedTxnAccessor.uncheckedFrom;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_ACCOUNT_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_QUERY_RANGE;
+import static com.hederahashgraph.api.proto.java.ResponseType.COST_ANSWER;
+
+public class GetAccountNftInfosAnswer implements AnswerService {
+    public static final String ACCOUNT_NFT_INFO_CTX_KEY = GetAccountNftInfosAnswer.class.getSimpleName() + "_accountNftInfos";
+    private OptionValidator validator;
+
+    public GetAccountNftInfosAnswer(OptionValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(Query query) {
+        return COST_ANSWER == query.getTokenGetAccountNftInfos().getHeader().getResponseType();
+    }
+
+    @Override
+    public boolean requiresNodePayment(Query query) {
+        return typicallyRequiresNodePayment(query.getTokenGetAccountNftInfos().getHeader().getResponseType());
+    }
+
+    @Override
+    public Response responseGiven(Query query, StateView view, ResponseCodeEnum validity, long cost) {
+        return responseFor(query, view, validity, cost, NO_QUERY_CTX);
+    }
+
+    @Override
+    public Response responseGiven(
+            Query query,
+            StateView view,
+            ResponseCodeEnum validity,
+            long cost,
+            Map<String, Object> queryCtx
+    ) {
+        return responseFor(query, view, validity, cost, Optional.of(queryCtx));
+    }
+
+    @Override
+    public ResponseCodeEnum checkValidity(Query query, StateView view) {
+        var accountNftInfoQuery = query.getTokenGetAccountNftInfos();
+        AccountID id = accountNftInfoQuery.getAccountID();
+
+        if (accountNftInfoQuery.getStart() > accountNftInfoQuery.getEnd()) {
+            return INVALID_QUERY_RANGE;
+        }
```

Comment:
I think `start` must be strictly less than `end` as specified in the protos. So here we want `>=` instead.
Also `INVALID_QUERY_RANGE` response code is not quite descriptive. We may want to modify it into `INVALID_NFT_INFOS_QUERY_RANGE` or something like that

Reply:
The idea behind `INVALID_QUERY_RANGE` not specifying NFTs was because we might want to reuse the same concept of range queries for other types. F.e querying account's relationships with tokens might have range in the future or any other type of query